drivers/soundwire/intel_init.c

```diff
@@ -219,9 +253,9 @@ static struct sdw_intel_ctx
 		return NULL;
 
 	ctx->count = count;
-	ctx->links = devm_kcalloc(&adev->dev, ctx->count,
-				  sizeof(*ctx->links), GFP_KERNEL);
-	if (!ctx->links)
+	ctx->ldev = devm_kcalloc(&adev->dev, ctx->count,
+				 sizeof(*ctx->ldev), GFP_KERNEL);
```

Comment:
@plbossart this seems wrong. You dont do a ancillary_device_add() until later in this function. So devm_ should fail isnt it? This would mean that you will need the release callback for the ancillary device as well.

Reply:
This is memory allocated by the ACPI companion of the PCI device. I can allocate all the memory I want here, regardless of what type of device I end-up creating.

adev here does not mean ancillary, it's ACPI! That's why I used 'ancildev' everywhere, to avoid the confusion.

Reply:
ahh yes. I got confused by the adev there.